util/cache/single.go

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package cache
+
+import (
+	"time"
+)
+
+// Single is a simple in-memory cache that stores a single value until a
+// defined time before it is re-fetched. It also supports returning a
+// previously-expired value if refreshing the value in the cache fails.
+//
+// Single is not safe for concurrent use.
```

Comment:
Not sure if this is a good idea; see what you think: It might be nice to provide a `type LockingCache[C Cache, ...] struct { ... }` that can lock any cache, and forwards method calls to the underlying `C`, holding a lock as necessary.

Then you could document just once to the effect of, "If you need a concurrency-safe cache, use `LockingCache`..."

Reply:
Good idea; just added that as an implementation.